rpcs3/Emu/system_utils.cpp

```diff
@@ -192,11 +283,13 @@ namespace rpcs3::utils
 
 		if (category == "HG" && !content_id.empty())
 		{
-			// This is a trial game. Check if the user has a RAP file to unlock it.
-			if (fs::is_file(game_path + "/C00/PARAM.SFO") && fs::is_file(get_rap_file_path(content_id)))
+			// This is a trial game. Check if the user has EDAT file to unlock it.
+			const auto c00_title_id = std::string(psf::get_string(psf, "TITLE_ID"));
```

Comment:
There's nothing wrong with string_view here.

Reply:
How would use it below then?

Reply:
use string view instead of const string ref.

Reply:
It won't let me, says "no suitable user-defined conversion from "const std::string_view" to "const std::string" exists".

Reply:
That's because you should use `std::string_view` instead of `const std::string&` in function arguments, that's what I meant.

Reply:
You're forgetting that `verify_c00_unlock_edat` is also called from `npDrmVerifyUpgradeLicense`.

Reply:
std::string is implicitly convertible to std::string_view, I don't see what you're implying.

Reply:
Why would I use std::string_view as function arguments if it's primary use is in sceNp.cpp which only uses std::string?

Reply:
But what I suggested in the review is of string_view  type.

Reply:
const auto c00_title_id = psf::get_string(psf, "TITLE_ID");

Reply:
Not to mention changing arguments to std::string_view also creates problems within the function as it calls `get_rap_file_path` which uses std::string as arguments as well and _that_ is called in a whole bunch of places all of which will break.

Reply:
Ok, I see that there's a misunderstanding of what explicit and implicit conversions in c++ mean.

Reply:
Edited comment*

Reply:
> 
> 
> Ok, I see that there's a misunderstanding of what explicit and implicit conversions in c++ mean.
> https://en.cppreference.com/w/c/language/conversion
> https://en.cppreference.com/w/cpp/language/explicit_cast
> std::string -> std::string_view: implcit.
> std::string_view -> std::string: explicit.

Yeah, I got what you meant after I sent my first post but the point about `get_rap_file_path` still stands.

Reply:
> 
> 
> Ok, I see that there's a misunderstanding of what explicit and implicit conversions in c++ mean.
> https://en.cppreference.com/w/c/language/conversion
> https://en.cppreference.com/w/cpp/language/explicit_cast
> std::string -> std::string_view: implcit.
> std::string_view -> std::string: explicit.

My bad, I got what you meant after I sent the first post. What about `get_rap_file_path', though?

Reply:
Change it to string_view as well.

Reply:
It calls `fmt::format`, are there are any issues with mixing std::string and std::string_view args there?

Reply:
There is no such issue.

Reply:
Got it.